### PR TITLE
Have divexact raise an exception when it isn't an exact division #282.

### LINKED ISSUE #282: Integer divexact doesn't raise exception (efficiently) if division is not exact
We can't use mpz_divisible_p in Flint, since this doesn't return the quotient. Nor can we use mpz_divexact, since this just assumes exact quotient. For now, we have to use division with remainder and explicitly check the remainder.

It would be more optimal to have a function that uses Hensel division, like mpz_divexact, but checks the Hensel remainder to determine if the division was exact. Unfortunately this function doesn't exist in GMP or MPIR, and hence not in Flint.

This is considerable work to fix, but is an important optimisation we'll need one day.

### LINKED ISSUE #698: divexact(ZZ(2), 3) doesn't raise exception

### PR DESCRIPTION
**Pull request by wbhart**
Closes #282 
Closes #698

### CODE REVIEW

**Comment by rfourquet** [comment]
Looks good. I suggest using `ArgumentError` instead of `ErrorException`, and to delete the method taking `Int` as it's redundant now. I prepared a commit for that that I wanted to push to your branch, but it doesn't work (I thought it would because of the message `Add more commits by pushing to the fix_divexact_ZZ branch on wbhart/Nemo.jl.` So I pushed the branch [fix_divexact_ZZ](https://github.com/Nemocas/Nemo.jl/tree/fix_divexact_ZZ) on the main repo if you like to incorporate the changes.

File: test/flint/fmpz-test.jl [hunk 1]
```diff
@@ -135,6 +135,7 @@ end
 
 @testset "fmpz.exact_division..." begin
    @test divexact(fmpz(24), fmpz(12)) == 2
+   @test_throws ErrorException divexact(fmpz(24), fmpz(11))
```
Comment (rfourquet) [inline test/flint/fmpz-test.jl]: You will have to change the exception here, same below.

**Comment by wbhart** [comment]
@rfourquet Is this ok now?

**Comment by wbhart** [comment]
@rfourquet Whoops, I clicked the wrong button and merged it. Let me know if there is anything I missed and I will add it in another PR.

**Comment by rfourquet** [comment]
No I merged it myself :)

**Comment by wbhart** [comment]
Ah, that explains it! Thanks.

**Pull request merged**